### DISCUSSION TITLE
Combining coldtype with other python library

### DISCUSSION THREAD

**Discussion by jama1017** [coldtype/coldtype#101] [Q&A] [answered]
Hi!

First of all, thanks so much for the awesome work on this library. I really love what you can do with it.

I am working on a project that aims to drive typography animation with video content (e.g. animate texts based on people's facial expressions in the video). I am new to this library, and, after installing it, I realized that you need to run the scripts with `coldtype test.py` instead of just `python test.py`. 

I wonder if there is a way to run coldtype scripts with `python test.py`, so that I can also use coldtype with other Python libraries together? Or, more broadly, how can I use this library with other libraries, like OpenCV and MoviePy?

**Comment by stenson** [comment, marked as answer]
Thanks for the kind words — sounds like a fun project! If you have your own rendering/viewing pipeline, it shouldn't be too hard to use coldtype in a "headless" fashion, either by generating bitmaps or vector data that you could then render on your own. (The idea behind the `coldtype` command-line utility is to remove boilerplate around that kind of thing from a coldtype source file.)

Here's an example of some code that could be run without the `coldtype` command-line program:

```python
import coldtype as ct
from coldtype.fx.skia import rasterize
from coldtype.pens.svgpen import SVGPen

rect = ct.Rect(1080, 1080)
txt = (ct.StSt("Hello World", "Times", 100)
    .f(0)
    .align(rect))

# as an image
txt.ch(rasterize(rect, "test.png"))

# as a vector
print(SVGPen.Composite(txt, rect, viewBox=False))
```

Alternatively I _think_ you could use those libraries from within a coldtype program (i.e. something with a `@renderable` or `@animation` decorator), though that would depend on what you're looking to accomplish. Happy to discuss further either here or via email (rob.stenson@gmail.com) if you have some more specifics about how you're using those libraries.

**Reply by jama1017** [reply]
Thanks so much! I just followed up with an email. Hope to discuss further there!
